### DISCUSSION TITLE
The Button command in ItemTemplate synchronizes the IsEnabled effect

### DISCUSSION THREAD

**Discussion by CodingOctocat** [CommunityToolkit/dotnet#721] [Q&A] [answered]
I have a `ComboBox`, its `ItemTemplate` has a `Button`, the command is binding to `FooAsync()` in VM, this `FooAsync` may take 3s for IO operation, I found that when I click one of the buttons, all the buttons in the `ComboBoxItem` become grayed out and disabled(for 3s). How can I fix this problem?

**Comment by kmgallahan** [comment]
If you want a single ComboBox item to be disabled during command execution then the UI framework you are using is responsible for how `ICommand` updates are handled, rather than this toolkit.

**Comment by Sergio0694** [comment, marked as answer]
Are all buttons binding to the same command? If so, the reason is probably that async commands have concurrency control enabled by default — ie. they're disabled while an operation is running. You can change this by using `[RelayCommand(AllowConcurrentExecutions = true)]` (or passing the corresponding option, if you're constructing the command manually). Of course, then it's up to you to disable the buttons when needed, with whatever logic you want (eg. per button, with some criteria you want) 🙂

**Reply by CodingOctocat** [reply]
`[RelayCommand(AllowConcurrentExecutions = true)]` solved my problem, but as far as I remember, I tried to do this, and I also used the manual management `IsEnabled`, neither worked, maybe, my memory is wrong, maybe I was using `[RelayCommand(AllowConcurrentExecutions = false)]`...😕

**Reply by CodingOctocat** [reply]
As a side note, just setting `AllowConcurrentExecutions = true` is not enough, it will make the Button still clickable at the command `IsRunning=true`, so I still need to manually add an `IsBusy` to my data class and bind it to Button's `IsEnabled`.
